s3/src/main/scala/akka/stream/alpakka/s3/settings.scala

```diff
@@ -229,6 +230,119 @@ object AccessStyle {
   def virtualHostAccessStyle: VirtualHostAccessStyle = VirtualHostAccessStyle
 }
 
+final class RetrySettings private (val maxRetries: Int,
+                                   val minBackoff: FiniteDuration,
+                                   val maxBackoff: FiniteDuration,
+                                   val randomFactor: Double) {
+
+  /** Java API */
+  def getMaxRetries: Int = maxRetries
+
+  /** Java API */
+  def getMinBackoff: JavaDuration = JavaDuration.ofNanos(minBackoff.toNanos)
+
+  /** Java API */
+  def getMaxBackoff: JavaDuration = JavaDuration.ofNanos(maxBackoff.toNanos)
+
+  /** Java API */
+  def getRandomFactor: Double = randomFactor
+
+  def withMaxRetries(value: Int): RetrySettings = copy(maxRetries = value)
+
+  def withMinBackoff(value: FiniteDuration): RetrySettings = copy(minBackoff = value)
+
+  def withMinBackoff(value: JavaDuration): RetrySettings =
```

Comment:
Can you decorate these java setters with `/** Java API */` too?

Reply:
Sure.  I just added it to the ones that take `java.time.Duration` but let me know if I should add it to the others too.

Reply:
I think that's all of them!